lib/solidus_gdpr/data_segments/base.rb

```diff
@@ -40,7 +45,8 @@ def export
       # Erases all data contained in this segment.
       #
       # This is used by the {DataEraser} service.
-      def erase
+      # @param erased_email [String] it will replace the current one
+      def erase(_erased_email)
```

Comment:
I'm not really convinced about this.

I don't think the gem itself should know about the details of how emails are erased by data segments. Keep in mind users might override the `profile` segment to not erase the user's email, and then you have a completely useless and confusing argument being passed to `erase`.

I'm not sure what a better option would look like right now, but let's see if we come up with any ideas.